### PR TITLE
Actually use ShapeInfo for Simplex::FE_P and Simplex::QGauss

### PR DESCRIPTION
**Pull request by peterrum**
... follow up to #11186.

### CODE REVIEW

File: include/deal.II/matrix_free/mapping_info.h [hunk 1]
```diff
@@ -125,6 +125,14 @@ namespace internal
          */
         QuadratureDescriptor();
 
+        /**
+         * Set up the lengths in the various members of this struct.
+         */
+        template <int dim_q>
+        void
+        initialize(const Quadrature<dim_q> &quadrature_1d,
```
Comment (drwells) [inline include/deal.II/matrix_free/mapping_info.h:133]:
```suggestion
        initialize(const Quadrature<dim_q> &quadrature,
```

**Review by drwells — COMMENTED** [review]
This has some changes from #11167. Which should be merged first?

File: include/deal.II/matrix_free/mapping_info.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/matrix_free/mapping_info.h]: You are right :)

**Comment by peterrum** [comment]
> Which should be merged first?

This one!

**Review by drwells — APPROVED** [review]

**Comment by drwells** [comment]
The parallel mac tester is failing for known reasons that are completely unrelated to this patch - I'm just going to go ahead and merge since everything else passed.

**Pull request merged**